sql/log_event.cc

```diff
@@ -10723,7 +10723,21 @@ int Rows_log_event::do_apply_event(Relay_log_info const *rli)
       DBUG_RETURN(-1);
     }
     else if (state == GTID_STATEMENT_SKIP)
+    {
```

Comment:
I am not sure about this. This question requires additional research and there indeed must be other test to model different situations(I counted at least four such cases) when gtid_pre_statement_checks() returns GTID_STATEMENT_CANCEL and how higher-level functions process the case when Rows_log_event::do_apply_event() returns error. I think such cases must be considered separately from the original case, and there must be separate bug report.